src/java/org/apache/cassandra/net/OutboundTcpConnection.java

```diff
@@ -387,6 +387,7 @@ private void disconnect()
             }
             out = null;
             socket = null;
+            poolReference.socketDisconnectListener.accept(this);
```

Comment:
Reading a bit more, this will be quite aggressive, but we really don't expect the resets to be happening too much, so cancelling overly aggressively, is possibly ok.

If not, we may need to track which requests have actually been sent vs are still in our queue, and maybe not cancel those we haven't sent.